.github/workflows/zero-downtime-tests.yml

```diff
@@ -1,13 +1,456 @@
-name: Zero-Downtime Tests
+name: ZeroDowntime Tests
 on:
-  workflow_dispatch:
+  workflow_dispatch: # run integration tests only when triggered manually
+    inputs:
+      branch:
+        description: 'Take CI build artifacts from branch (e.g., master, release-x.y.z)'
+        required: true
+        default: 'master'
+      upgradeTo:
+        description: 'Helm chart path to run upgrade command'
+        required: true
+        default: '0.15.0-dev-PR-7504'
+      deleteOnFinish:
+        type: boolean
+        required: false
+        description: "Check this if you don't want the test namespaces to stay alive after the test run"
+        default: false
 defaults:
   run:
     shell: bash
 jobs:
-  hello-world:
-    name: "Hello World"
+  calculate-queue-time:
+    name: "Calculate Queue Time"
+    uses: ./.github/workflows/pipeline-queuetime.yml
+    with:
+      workflow_name: "Zero Downtime Tests"
+    secrets:
+      monitoring_api_token: ${{ secrets.MONITORING_API_TOKEN }}
+      monitoring_tenant_url: ${{ secrets.MONITORING_TENANT_URL }}
+
+  zd-test:
+    name: "Tests"
     runs-on: ubuntu-20.04
+    needs: calculate-queue-time
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - CLOUD_PROVIDER: "GKE"
+            PLATFORM_VERSION: "1.21"
+            CLUSTER_NAME: "keptn-integration-test-2"
+            KUBECONFIG: ""
+            PLATFORM: "kubernetes"
+            KEPTN_SERVICE_TYPE: "LoadBalancer"
+            COLLECT_RESOURCE_LIMITS: "true"
+            USE_RESOURCE_SERVICE: "true"
+    env:
+      CLOUD_PROVIDER: ${{ matrix.CLOUD_PROVIDER }}
+      PLATFORM: ${{ matrix.PLATFORM }}
+      PLATFORM_VERSION: ${{ matrix.PLATFORM_VERSION }}
+      KUBECONFIG: ${{ matrix.KUBECONFIG }}
+      KEPTN_NAMESPACE: "keptn-zd-test-${{ github.run_number }}-${{ github.run_attempt }}"
+      KEPTN_SERVICE_TYPE: ${{ matrix.KEPTN_SERVICE_TYPE }}
+      COLLECT_RESOURCE_LIMITS: ${{ matrix.COLLECT_RESOURCE_LIMITS }}
+      GO_VERSION: 1.17
+      TEST_REPORT_FOLDER: test-reports-${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}
+      FINAL_TEST_REPORT_FOLDER: test-reports
+      FINAL_TEST_REPORT_PATH: test-reports/test-report-final-${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}.log
+      TEST_REPORT_FILENAME_SUFFIX: test-report-${{ github.run_id }}-${{ matrix.PLATFORM }}-${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}.log
```

Comment:
since test reporting doesn't happen in this pipeline, all those variables can be removed i think